discord-scripts/invite-management.ts

```diff
@@ -0,0 +1,181 @@
+    }
+    // create an invite based of the command and channel where the command has been run
+    discordClient.on("interactionCreate", async (interaction) => {
+      if (
+        !interaction.isCommand() ||
+        interaction.commandName !== "create-invite"
+      )
+        return
+
+      if (!interaction.guild) {
+        await interaction.reply("This command can only be used in a server.")
+        return
+      }
+
+      const maxAge =
+        ((interaction.options.get("max-age", false)?.value as number) ?? 24) *
+        3600
+      const maxUses =
+        (interaction.options.get("max-uses", false)?.value as number) ?? 10
+
+      try {
+        const { channel } = interaction
+        if (channel instanceof TextChannel) {
+          const invite = await channel.createInvite({
+            maxAge,
+            maxUses,
+            unique: true,
+          })
+          await interaction.reply(
+            `Here is your invite link: ${invite.url}\nThis invite expires in ${
+              maxAge / 3600
+            } hours and has a maximum of ${maxUses} uses.`,
+          )
+        } else {
+          await interaction.reply(
+            "Cannot create an invite for this type of channel.",
+          )
+        }
+      } catch (error) {
+        await interaction.reply("An error occurred while creating an invite.")
+      }
+    })
+
+    // Generates an invite if the channel name matches ext-*-audit format
+    discordClient.on("channelCreate", async (channel) => {
+      if (channel.parent && channel.parent.name === "defense") {
+        const regex = /^ext-.*-audit$/
+        if (regex.test(channel.name)) {
+          try {
+            const channelId = channel.id
+            const maxAge = 86400 // 1 day
+            const maxUses = 5
+            const inviteUrl = await createInvite(
+              discordClient,
+              channelId,
+              maxAge,
+              maxUses,
+            )
+            if (inviteUrl) {
+              robot.logger.info(
+                `New invite created for defense audit channel: ${channel.name}, URL: ${inviteUrl}`,
+              )
+              if (channel instanceof TextChannel) {
+                channel.send(
+                  `Here is your invite link: ${inviteUrl}\n` +
+                    `This invite expires in ${
+                      maxAge / 3600
+                    } hours and has a maximum of ${maxUses} uses.`,
+                )
+              }
+            }
+            // create a new role with the exact name of the channel with permissions only to that channel
+            const role = await channel.guild.roles.create({
+              name: channel.name,
+              reason: `Role for ${channel.name} channel`,
+            })
+
+            await channel.permissionOverwrites.create(role, {
+              ViewChannel: true,
+              SendMessages: true,
+              ReadMessageHistory: true,
```

Comment:
I would nuke these, I believe the only permission overwrite we need in our matrix is `ViewChannel` by design.